traces_profiler.go

```diff
@@ -0,0 +1,70 @@
+package sentry
+
+func (span *Span) startProfiling() {
+	var sampleRate = span.clientOptions().ProfilesSampleRate
+	switch {
+	case sampleRate < 0.0 || sampleRate > 1.0:
+		Logger.Printf("Skipping transaction profiling: ProfilesSampleRate out of range [0.0, 1.0]: %f", sampleRate)
+	case sampleRate == 0.0 || rng.Float64() >= sampleRate:
+		Logger.Printf("Skipping transaction profiling: ProfilesSampleRate is: %f", sampleRate)
+	default:
+		span.profiler = &_transactionProfiler{
+			stopFunc: startProfiling(),
+		}
+	}
+}
+
+type transactionProfiler interface {
+	Finish(span *Span, event *Event) *profileInfo
+}
+
+type _transactionProfiler struct {
+	stopFunc func() *profilerResult
+}
+
+// Finish implements transactionProfiler.
+func (tp *_transactionProfiler) Finish(span *Span, event *Event) *profileInfo {
+	result := tp.stopFunc()
+	info := &profileInfo{
+		Version:     "1",
+		Environment: event.Environment,
+		EventID:     uuid(),
+		Platform:    "go",
+		Release:     event.Release,
```

Comment:
We added `dist` as well (https://github.com/getsentry/relay/blob/master/relay-profiling/src/sample.rs#L158).

Reply:
thanks for the headsup